torch/_dynamo/trace_rules.py

```diff
@@ -92,7 +92,7 @@
     "torch.distributed.is_initialized": TorchInGraphFunctionVariable,
     "torch.distributed.get_rank": TorchInGraphFunctionVariable,
     "torch.distributed.get_world_size": TorchInGraphFunctionVariable,
-    "torch.distributed._tensor.DTensor#from_local": TorchInGraphFunctionVariable,
+    "torch.distributed._tensor.api.DTensor#from_local": TorchInGraphFunctionVariable,
```

Comment:
Should we put both of them here? It seems they are not identical, so probably others use the previous one, so we should put them in graph for both case?
Actually inside of the variable, we are using the previous one:
https://github.com/pytorch/pytorch/blob/34a61c527b242e13d617f853dcaa93d41b852320/torch/_dynamo/variables/distributed.py#L40-L45

It seems we have to cover both cases if they are not identical.

Reply:
IMO they should be identical objects, it's just that `torch.distributed._tensor.DTensor` is imported from `torch.distributed._tensor.api.DTensor`.  code [link](https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/__init__.py#L9)

But indeed in some cases it does not hit the trace name map... which is really weird, it would be good if we could follow up what happened there.

If you want to be extra safe I can add both to the list, but I think the current change should capture the previous case as well, lmk :)

Reply:
SG!

Reply:
cool I think I'll land this as is to avoid trigger CI again, we can always add the previous one later if we hit problems.